compiler/rustc_ty_utils/src/consts.rs

```diff
@@ -222,6 +222,9 @@ impl<'a, 'tcx> AbstractConstBuilder<'a, 'tcx> {
     fn recurse_build(&mut self, node: thir::ExprId) -> Result<NodeId, ErrorGuaranteed> {
         use thir::ExprKind;
         let node = &self.body.exprs[node];
+        let param_env = ParamEnv::reveal_all().with_reveal_all_normalized(self.tcx);
+        let node_ty = self.tcx.try_normalize_erasing_regions(param_env, node.ty).unwrap_or(node.ty);
```

Comment:
Both `reveal_all` and `normalize_erasing_regions` are almost always the wrong thing. Unless you know what it means to use it, you can be sure it's wrong to use it.

The fact that an unnormalized type ends up in a `node.ty` is the problem, find out where it comes from and normalize there.

Reply:
Ah I marked this as resolved mistakenly. Should still check where the node gets its type and normalize there